clients/python/src/model_registry/_async_task_runner.py

```diff
@@ -0,0 +1,102 @@
+# from https://gist.github.com/blink1073/969aeba85f32c285235750626f2eadd8
```

Comment:
We should verify if alternative libraries exists on pypi to avoid having to introduce multiple licensing in the project itseld

Reply:
@tarilabs, We had the same need in our internal project, and this was the only solution that I found after more than a week of searches and conversations with my colleagues. If you can find a better solution for this I will be very interested to hear it

Reply:
Sure, but maybe there are ways we could allow user to plug their own executor/runner strategies, without requiring necessarily to opt for one or the other ourselves, maybe picking one

(this is because seems to me at the end of the day it's the same signature/contract)

Reply:
Do you mean introducing a base class and using AsyncTaskRunner as a default? To make ModelRegistry independent of it we will also need a factory class, doing instantiation.

So add 2 additional classes. So yes, this can be done. Is it worth it?
If the answer is yes, we can do it